File: pwdmg_core/paths.py
from __future__ import annotations

from pathlib import Path

APP_NAME = "My Password Manager"
APP_DIR = Path.home() / "mypwdmg"
VAULT_FILE = APP_DIR / "vault.json"
LEGACY_LOCAL_STORAGE_FILE = APP_DIR / "localStorage_data.json"
DESKTOP_CONFIG_FILE = APP_DIR / "desktop_config.json"
LOCAL_BACKUP_DIR = APP_DIR / "backups"
UPDATE_DIR = APP_DIR / "updates"
NATIVE_HOST_DIR = APP_DIR / "native-host"
PLUGIN_CONFIG_FILE = APP_DIR / "plugin_config.json"
DEVICE_UNLOCK_FILE = APP_DIR / "device_unlock.json"
ATTACHMENT_DIR = APP_DIR / "attachments"


DEFAULT_DESKTOP_CONFIG = {
    "appname": "My Password Manager",
    "width": 360,
    "height": 480,
    "x_position": 160,
    "y_position": 80,
    "tray_enabled": True,
    "close_behavior": "minimize-to-tray",
}


def ensure_app_dir() -> Path:
    APP_DIR.mkdir(parents=True, exist_ok=True)
    return APP_DIR

File: src-tauri/crates/core/src/paths.rs
use std::path::PathBuf;

pub const APP_NAME: &str = "My Password Manager";

pub fn app_dir() -> PathBuf {
    directories::UserDirs::new()
        .map(|u| u.home_dir().to_path_buf())
        .unwrap_or_else(|| PathBuf::from("."))
        .join("mypwdmg")
}

pub fn ensure_app_dir() -> std::io::Result<PathBuf> {
    let dir = app_dir();
    std::fs::create_dir_all(&dir)?;
    Ok(dir)
}

pub fn vault_file() -> PathBuf {
    app_dir().join("vault.json")
}

pub fn legacy_local_storage_file() -> PathBuf {
    app_dir().join("localStorage_data.json")
}

pub fn desktop_config_file() -> PathBuf {
    app_dir().join("desktop_config.json")
}

pub fn local_backup_dir() -> PathBuf {
    app_dir().join("backups")
}

pub fn update_dir() -> PathBuf {
    app_dir().join("updates")
}

pub fn native_host_dir() -> PathBuf {
    app_dir().join("native-host")
}

pub fn plugin_config_file() -> PathBuf {
    app_dir().join("plugin_config.json")
}

pub fn device_unlock_file() -> PathBuf {
    app_dir().join("device_unlock.json")
}

pub fn attachment_dir() -> PathBuf {
    app_dir().join("attachments")
}
